Makefile.pre.in

```diff
@@ -1994,6 +1994,24 @@ testuniversal: all
 	$(RUNSHARED) /usr/libexec/oah/translate \
 		./$(BUILDPYTHON) -E -m test -j 0 -u all $(TESTOPTS)
 
+# Run the test suite on the iOS simulator. Must be run on a macOS machine with
+# a full Xcode install that has an iPhone SE (3rd edition) simulator available.
+XCFOLDER=$(srcdir)/build/iOSTestbed.$(MULTIARCH).$(shell date +%s)
+XCRESULT=$(XCFOLDER)/$(MULTIARCH).xcresult
+.PHONY: testios
+testios: install
+	# Copy the testbed project into the build folder
+	cp -r $(srcdir)/iOS/testbed $(XCFOLDER)
+	# Copy the framework from the install location to the testbed project.
+	cp -r $(PYTHONFRAMEWORKPREFIX)/* $(XCFOLDER)/Python.xcframework/ios-arm64_x86_64-simulator
+	# Run the test suite for the Xcode project, targeting the iOS simulator.
+	# If the suite fails, extract and print the console output, then re-raise the failure
+	if ! xcodebuild test -project $(XCFOLDER)/iOSTestbed.xcodeproj -scheme "iOSTestbed" -destination "platform=iOS Simulator,name=iPhone SE (3rd Generation)" -resultBundlePath $(XCRESULT) ; then \
+		xcrun xcresulttool get --path $(XCRESULT) --id $$(xcrun xcresulttool get --path $(XCRESULT) --format json | $(PYTHON_FOR_BUILD) -c "import sys, json; result = json.load(sys.stdin); print(result['actions']['_values'][0]['actionResult']['logRef']['id']['_value'])"); \
+		echo ; \
```

Comment:
One other builder-friendliness issue here is that the name of the framework is hardwired to `Python` throughout the testbed, including the `.xcodeproj` metadata. I suppose there are various ways that we could handle it, like making all the relevant files `configure` files or possibly adding Xcode-supported variables but that seems (even to me!) like a lot of work for little benefit. At a minimum, I suppose we could add a test at the start of the `testios` rule to see if the framework name was changed by `--with-framework-name` and error in that case.

BTW, since I haven't figured out an easy way to make a comment in this PR at the proper line in `configure.ac`, I failed earlier to test using a different name and have now found that the iOS definitions for LDSHARED and LDCXXSHARED have a hardwired framework name and should be changed:

```
 	iOS/*)
-		LDSHARED='$(CC) -dynamiclib -F . -framework Python'
-		LDCXXSHARED='$(CXX) -dynamiclib -F . -framework Python'
+		LDSHARED='$(CC) -dynamiclib -F . -framework $(PYTHONFRAMEWORK)'
+		LDCXXSHARED='$(CXX) -dynamiclib -F . -framework $(PYTHONFRAMEWORK)'
 		BLDSHARED="$LDSHARED"

```

Reply:
Agreed the hard-coded framework name is an issue here. I've updated the code to use link the right name in `LDSHARED` et al, and raise an error if you try to run `testios` with a custom framework name; I've also added a note in the docs warning about the problem.